tests/pipeline/test_events.py

```diff
@@ -1,299 +1,425 @@
-        super(EventDataSetLoader, self).__init__(
-            all_dates,
-            events_by_sid,
-            infer_timestamps=infer_timestamps,
-            dataset=dataset,
-        )
+    previous_string_custom_missing = Column(
+        dtype=categorical_dtype,
+        missing_value=u"<<NULL>>",
+    )
+    next_string_custom_missing = Column(
+        dtype=categorical_dtype,
+        missing_value=u"<<NULL>>",
+    )
 
-    @lazyval
-    def previous_announcement_loader(self):
-        return self._previous_event_date_loader(
-            self.dataset.previous_announcement,
-        )
 
-    @lazyval
-    def next_announcement_loader(self):
-        return self._next_event_date_loader(
-            self.dataset.next_announcement,
-        )
+def make_events_for_sid(sid, event_dates, event_timestamps):
+    num_events = len(event_dates)
+    return pd.DataFrame({
+        'sid': np.full(num_events, sid, dtype=np.int64),
+        'timestamp': event_timestamps,
+        'event_date': event_dates,
+        'float': np.arange(num_events, dtype=np.float64) + sid,
+        'int': np.arange(num_events) + sid,
+        'datetime': pd.date_range('1990-01-01', periods=num_events).shift(sid),
+        'string': ['-'.join([str(sid), str(i)]) for i in range(num_events)],
+    })
 
 
-# Test case just for catching an error when multiple columns are in the wrong
-#  data format, so no loader defined.
-class EventDataSetLoaderMultipleExpectedColsNoColumnLoaders(EventsLoader):
-    expected_cols = frozenset([ANNOUNCEMENT_FIELD_NAME, OTHER_FIELD])
+critical_dates = pd.to_datetime([
+    '2014-01-05',
+    '2014-01-10',
+    '2014-01-15',
+    '2014-01-20',
+])
 
-    event_date_col = ANNOUNCEMENT_FIELD_NAME
 
+def make_events():
+    """
+    Every event has at least three pieces of data associated with it:
 
-class EventDataSetLoaderNoExpectedCols(EventsLoader):
+    1. sid : The ID of the asset associated with the event.
+    2. event_date : The date on which an event occurred.
+    3. timestamp : The date on which we learned about the event.
+                   This can be before the occurence_date in the case of an
+                   announcement about an upcoming event.
 
-    def __init__(self,
-                 all_dates,
-                 events_by_sid,
-                 infer_timestamps=False,
-                 dataset=EventDataSet):
-        super(EventDataSetLoaderNoExpectedCols, self).__init__(
-            all_dates,
-            events_by_sid,
-            infer_timestamps=infer_timestamps,
-            dataset=dataset,
-        )
+    Events for two different sids shouldn't interact in any way, so the
+    interesting cases are determined by the possible interleavings of
+    event_date and timestamp.
+
+    Fix two events with dates e1, e2 and timestamps t1 and t2.
+
+    Without loss of generality, assume that e1 < e2. (If two events have the
+    same occurrence date, the behavior of next/previous event is undefined).
```

Comment:
I don't think there's a way that we can deterministically choose an event if the events have the same date.  If the timestamps are different, we could make an arbitrary choice to take the event with the newer and/or older timestamp, but if both the event date and timestamp are the same, there's no way for us to distinguish the events here.